pkg/types/rekord/v0.0.1/rekord_v0_0_1_schema.json

```diff
@@ -86,7 +86,8 @@
                 "content": {
                     "description": "Specifies the content inline within the document",
                     "type": "string",
-                    "format": "byte"
+                    "format": "byte",
+                    "maxLength" : 32000000
```

Comment:
I guess we can't comment here in the JSON. I'll make a note in the nginx config to remember to update both places.

Reply:
OK, filed a TODO over there.

Reply:
While this change would limit the length of the inline base64 encoded content field, it's not sufficient to develop an equivalent limit to what the NGINX ingress was enforcing (i.e. the entire number of bytes passed in the HTTP request body). Also since OpenAPI is enforcing this via the length of the string, the 32M value really puts a limit on the content of ~67% of that or approx 21MB. 

I can see use cases for passing large files to Rekor (e.g. signed ISO images or jumbo containers), and since certain types can be fully streamed (rekord and RPM) while others can't (soon-to-be-added JAR support), I think we need to  revert this PR and consider it more broadly. I'm happy to take some time to develop a proposal on this. 

I could also imagine some users might want a size limit to be configurable, so that is something to consider as well.

Reply:
> I could also imagine some users might want a size limit to be configurable, so that is something to consider as well.

Agree, I think this might have been suitable being a config value in rekor-server.yaml, that way you can place `32000000` as the fault value in the config, but users of rekor who have needs for larger file can still set accordingly.

Reply:
That sounds great, would love to chat more about office hours! I had exactly that problem with not being able to measure the request itself, and couldn't see a great solution here.

Reply:
The issue right now is that it's enforced at ingress, not in our rekor config. That's the main reason client-validation would be nice, otherwise we end up with random errors from the load balancer that clients can't catch.

It might be better to just let clients hit it though.

Reply:
We could also return back the 413 w/ error message